server/src/main/scala/org/http4s/server/middleware/authentication/DigestUtil.scala

```diff
@@ -30,6 +30,43 @@ private[authentication] object DigestUtil {
   private def md5[F[_]: Monad: Hash](str: String): F[String] =
     Hash[F].digest(HashAlgorithm.MD5, ByteVector.view(str.getBytes())).map(_.toHex)
 
+  /** Computes the response value used in Digest Authentication.
+    * @param method
+    * @param ha1
+    * @param uri
+    * @param nonce
+    * @param nc
+    * @param cnonce
+    * @param qop
+    * @return
+    */
+  def computeHashedResponse[F[_]: Monad: Hash](
+      method: String,
+      ha1: String,
+      uri: String,
```

Comment:
Is this the path or the full URI?  I haven't connected the dots.  There are case-insensitive bits to a full URI.  I'm wondering what the spec requires here.

Reply:
This is `req.uri.toString()` -- I don't see anything in the spec that explicitly clarifies uri being case sensitive or insensitive.

That being said, considering that http permits different content to be at different uris which could be equal when compared case-insensitively, I think it follows that this should also be case-sensitive.

I could push `Uri` down into the `uri` parameter, if desired.

Reply:
That might give us flexibility in case we get it wrong.  This just feels like the kind of issue that bites us in the ass later.

> The URI from Request-URI of the Request-Line; duplicated here because proxies are allowed to change the Request-Line in transit.

I don't see anything about normalizing it, so what you've done is _probably_ right.  But we could call `toString` here just as surely as we called `toString` wherever you called it.

Reply:
> But we could call toString here just as surely as we called toString wherever you called it.

I'll not take credit for what already existed before, I just broke up a single function that took all the parameters at the same time to a few that take different combinations of the parameters.

I'll make the change.